google/cloud/pubsub/internal/subscription_lease_management_test.cc

```diff
@@ -0,0 +1,201 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/pubsub/internal/subscription_lease_management.h"
+#include "google/cloud/pubsub/testing/mock_subscriber_stub.h"
+#include "google/cloud/internal/background_threads_impl.h"
+#include "google/cloud/testing_util/status_matchers.h"
+#include <gmock/gmock.h>
+#include <vector>
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+inline namespace GOOGLE_CLOUD_CPP_PUBSUB_NS {
+namespace {
+
+using ::google::cloud::testing_util::StatusIs;
+using ::testing::ElementsAre;
+
+google::pubsub::v1::PullResponse MakePullResponse(std::string const& prefix,
+                                                  int count) {
+  google::pubsub::v1::PullResponse response;
+  for (int i = 0; i != count; ++i) {
+    auto const id = prefix + std::to_string(i);
+    auto& m = *response.add_received_messages();
+    m.set_ack_id("ack-" + id);
+    m.mutable_message()->set_message_id("message-" + id);
+  }
+  return response;
+}
+
+TEST(SubscriptionLeaseManagementTest, NormalLifecycle) {
+  auto stub = std::make_shared<pubsub_testing::MockSubscriberStub>();
+
+  auto constexpr kTestDeadlineSeconds = 345;
```

Comment:
Consider making these `std::chrono::seconds`.

Reply:
Done, but I am not sure it is a win... I am happy either way.

Reply:
Undone. MSVC did not like the 64-bit integer -> double conversions.

Reply:
I'm not sure why `double` needs enter the picture, but OK.

[The win, I'd say, is to never be dealing with times/durations as raw integers, except right at an interface boundary where it is forced upon you.]